serial_protocol.md

```diff
@@ -186,3 +192,143 @@ Used to stop listening to events at a given path. This must be the same path pro
 	<<+OK
+
+#Messaging Protocol
+
+## BEGIN_MSG 
+Called to start communicating with Firebase Cloud Messaging, it requires a server key. You can also optionally supply any of the arguments listed on the MSG command to make them the defaults for this session. If you set them as defaults you don't need to specify them when calling MSG.
+
+###Usage
+	BEGIN_MSG %Server_Key% %Argument% %Argument% ...
+###Response
+	+OK - Accepted initialization parameters
+	-ERROR_MISSING_SERVER_KEY
+	-ERROR_PARSING_ARGUMENTS
+###Examples
+	>> BEGIN_MSG AIzaSyCk4GtbBa-XMQbc5TyT5oe1KnH71M-2lAZ
+	<< +OK
+	>> BEGIN_MSG AIzaSyCk4GtbBa-XMQbc5TyT5oe1KnH71M-2lAZ topic=iot_updates high_priority=true time_to_live=3600
+	<< +OK
+	>> BEGIN_MSG 
+	<< +ERROR_MISSING_SERVER_KEY
+	>> BEGIN_MSG AIzaSyCk4GtbBa-XMQbc5TyT5oe1KnH71M-2lAZ topic = iot_updates
+	<< +ERROR_PARSING_ARGUMENTS // Can't have spaces in arguments.
+
+## MSG
+Called to start composing a message. It is followed by a list of named arguments, all of these arguments may be listed on BEGIN_MSG as well to set them as defaults for the session. Even if they are listed on BEGIN_MSG they can be overriden when calling MSG.
+
+Named arguments are specified by " %name%=%value% " where neither names nor values will have spaces in them.
+
+### Arguments
+#### registration_ids
+This is a comma separated list of device registrations you wish to send the message to. The list can have between 1 and 100 ids in it so you can send to a single device this way too.
+##### Examples
+registration_ids=fQCLfBOGdh0...9k0,fQCLfBOGdh0...5j1
+
+#### registration_ids
+This is a comma separated list of device registrations you wish to send the message to. The list can have between 1 and 100 ids in it so you can send to a single device this way too.
+##### Examples
+registration_ids=fQCLfBOGdh0...9k0,fQCLfBOGdh0...5j1
+
+#### topic
+This is the name of a topic you wish to send the message to.
+##### Examples
+topic=news
+
+#### collapse_key
+Sets a collapse key on the message. When the server recieves a message with a collapse key it will discard any pending messages it has with the same key. So the client will only see the most recent one when it checks in.
+##### Examples
+collapse_key=LatestUpdate
+
+#### high_priority
+Can be "true" or "false". If set to true the user's device will wake up to receive the message. This is used when the message is high priority but a cost it paid in battery life for the user.
+
+##### Examples
+high_priority=true
+
+#### delay_while_idle
+Can be "true" or "false". If set to true the message will not be delivered until the user's device is active. This is used when the message can wait to be seen until the user is available.
+
+##### Examples
+delay_while_idle=false
+	
+#### time_to_live
+The message will expire after this amount of time in seconds. The default (and max value) is four weeks. Can be between 0 and 2,419,200.
+
+##### Examples
+time_to_live=500
+
+##Usage
+	MSG %Argument% %Argument% ...
```

Comment:
do you think a single line variant would be usefu? (maybe we can distinguish with suffix like we did originally for db).

Something like:
`MSG "TITLE" "INLINE_BODY" [data_key=data_value ...]`